pandas/types/cast.py

```diff
@@ -133,7 +133,8 @@ def trans(x):  # noqa
                 if dtype.tz:
                     # convert to datetime and change timezone
                     from pandas import to_datetime
-                    result = to_datetime(result).tz_localize(dtype.tz)
+                    result = to_datetime(result).tz_localize('utc')
```

Comment:
can add addtl tests in ``pandas/tests/types/test_cast.py`` fit them in as appropriate. take out this fix temporarily to validate your test cases, then add back to make sure they work.